scripts/check-schema-and-codegen/utils.ts

```diff
@@ -0,0 +1,88 @@
+import dns from "dns";
+import fs from "fs";
+import os from "os";
+import { generate } from "@graphql-codegen/cli";
+import { execSync } from "child_process";
+import process from "process";
+import { getConfig } from "../../codegen";
+
+const GITHUB_API = "https://api.github.com";
+const GQL_DIR = "graphql/schema";
+const LOCAL_SCHEMA = "sdlschema";
+const REPO = "/repos/evergreen-ci/evergreen";
+
+/**
+ * Checks if a given domain can be resolved.
+ * @async
+ * @param domain - The domain name to check.
+ * @returns - Resolves to `true` if the domain can be resolved, `false` otherwise.
+ */
+export const canResolveDNS = (domain: string) =>
+  new Promise((resolve) => {
+    dns.lookup(domain, (err) => {
+      if (err) {
+        resolve(false);
+      } else {
+        resolve(true);
+      }
+    });
+  });
```

Comment:
Is this necessary? We can avoid this request by just checking if the fetch request returned the same error.
<img width="568" alt="image" src="https://github.com/evergreen-ci/spruce/assets/4605522/adb25a7b-252a-4cb4-80a2-181ccfaef297">